bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FolderWatcherHandler.java

```diff
@@ -0,0 +1,257 @@
+                + File.separator + thing.getUID().getAsString().replace(':', '_') + ".data");
+        try {
+            previousFtpListing = WatcherCommon.initStorage(currentFtpListingFile);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            logger.debug("Can't write file {}, error message {}", currentFtpListingFile, e.getMessage());
+            return;
+        }
+
+        initJob = scheduler.scheduleWithFixedDelay(this::connectionKeepAlive, 0, config.pollInterval, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        if (executionJob != null) {
+            executionJob.cancel(true);
+        }
+        if (initJob != null) {
+            initJob.cancel(true);
+        }
+        if (ftp.isConnected()) {
+            try {
+                ftp.logout();
+                ftp.disconnect();
+            } catch (IOException e) {
+                logger.debug("Error terminating FTP connection: ", e);
+            }
+        }
+    }
+
+    private List<String> listDirectory(@Nullable FTPClient ftpClient, String parentDir, String currentDir, int level,
+            boolean recursive) throws IOException {
+        Instant dateNow = Instant.now();
+        List<String> dirList = new ArrayList<>();
+        final String dirToList = parentDir + ((!currentDir.equals("")) ? ("/" + currentDir) : "");
+
+        for (FTPFile file : ftpClient.listFiles(dirToList)) {
+            String currentFileName = file.getName();
+            if (currentFileName.equals(".") || currentFileName.equals("..")) {
+                continue;
+            }
+            if (file.isDirectory()) {
+                if (recursive) {
+                    try {
+                        dirList.addAll(listDirectory(ftpClient, dirToList, currentFileName, level + 1, recursive));
+                    } catch (IOException e) {
+                        logger.debug("Can't read FTP directory: {}", dirToList, e);
+                    }
+                }
+            } else {
+                long diff = ChronoUnit.HOURS.between(file.getTimestamp().toInstant(), dateNow);
+
+                if (diff < config.diffHours) {
+                    file.setName(dirToList + "/" + currentFileName);
+                    dirList.add("ftp:/" + ftpClient.getRemoteAddress().toString() + file.getName());
+                }
+            }
+        }
+        return dirList;
+    }
+
+    private void connectionKeepAlive() {
+        if (ftp == null || ftp.isConnected() == false) {
+            if (config.secureMode.equals("NONE")) {
+                ftp = new FTPClient();
+            } else {
+                if (config.secureMode.equals("IMPLICIT")) {
+                    ftp = new FTPSClient(true);
+                } else if (config.secureMode.equals("EXPLICIT")) {
+                    ftp = new FTPSClient(false);
+                } else {
```

Comment:
Please use a switch statement here:
```suggestion
            switch (config.secureMode) {
                case "NONE":
                    ftp = new FTPClient();
                    break;
                case "IMPLICIT":
                    ftp = new FTPSClient(true);
                    break;
                case "EXPLICIT":
                    ftp = new FTPSClient(false);
                    break;
```